src/fluree/db/json_ld/commit.cljc

```diff
@@ -35,46 +35,57 @@
         (vswap! iri-map assoc sid iri)
         iri))))
 
-(defn get-ref-iris
-  "Returns a list of object IRIs from a set of flakes.
-  Only to be used for ref? predicates"
-  [db iri-map compact-fn flakes]
+(defn- subject-block-pred
+  [db iri-map compact-fn list? p-flakes]
   (go-try
-    (loop [[flake & r] flakes
-           acc []]
-      (if flake
-        (recur r (conj acc (<? (get-s-iri (flake/o flake) db iri-map compact-fn))))
-        acc))))
+    (loop [[p-flake & r'] p-flakes
+           all-refs? nil
+           acc'      nil]
+      (let [pdt       (flake/dt p-flake)
+            ref?      (= const/$xsd:anyURI pdt)
+            [obj all-refs?] (if ref?
+                              [{"@id" (<? (get-s-iri (flake/o p-flake)
+                                                     db iri-map
+                                                     compact-fn))}
+                               (if (nil? all-refs?) true all-refs?)]
```

Comment:
It took me a bit of thinking to figure out what this condition was actually doing: if `all-refs?` is `nil` or `true`, set it to `true`, otherwise leave it unchanged.

I think it might be clearer to initialize it to `true` and then just pass it unchanged through the `ref?` branch.

```clojure
(if ref?
  [{"@id" <>} all-refs?]
  [{"@value" <>} false])
```

This is code golfing for future clarity, though - I was able to figure it out!

Reply:
The `nil` possibility is to represent the uninitialized state. This is because `false` should be a trap-door value when/if set (because one `false` should cause the fn to return `false`). I think I still prefer starting with `nil` over `true` because it accurately reflects that "we don't know yet" state of reality at the start of the loop. Not to say it couldn't be made clearer, however.